decidim-core/app/controllers/decidim/participatory_processes_controller.rb

```diff
@@ -6,15 +6,14 @@ module Decidim
   # A controller that holds the logic to show ParticipatoryProcesses in a
   # public layout.
   class ParticipatoryProcessesController < ApplicationController
-    include NeedsParticipatoryProcess
-
     layout "layouts/decidim/participatory_process", only: [:show]
 
-    skip_after_action :verify_participatory_process, only: [:index]
+    before_action -> { extend(NeedsParticipatoryProcess) }, only: [:show]
```

Comment:
I booted the app in production mode and did some testing via the following small patch.

```diff
   class ParticipatoryProcessesController < ApplicationController
     layout "layouts/decidim/participatory_process", only: [:show]
 
-    before_action -> { extend(NeedsParticipatoryProcess) }, only: [:show]
+    before_action -> do
+      puts "In index: #{respond_to?(:current_participatory_process)}"
+    end, only: [:index]
+
+    before_action -> do
+      puts "In show before extending: #{respond_to?(:current_participatory_process)}"
+      extend(NeedsParticipatoryProcess)
+      puts "In show after extending: #{respond_to?(:current_participatory_process)}"
+    end, only: [:show]
 
     helper Decidim::AttachmentsHelper
     helper Decidim::ParticipatoryProcessHelper

```

 The idea is to confirm that the functionally added to the class via `extend(NeedsParticipatoryProcess)` doesn't affect the real controller class, and thus it doesn't affect future controller instances.

When I load a participatory process page, I see that the controller is correctly enhanced with the `current_participatory_process` helper, but when I subsequently load the participatory processes index page, the `current_participatory_process` helper is no longer available.

It _seems_ to work as I expect.